src/feature/sift.cc

```diff
@@ -793,14 +793,13 @@ bool CreateSiftGPUExtractor(const SiftExtractionOptions& options,
   sift_gpu_args.push_back("-v");
   sift_gpu_args.push_back("0");
 
-  // Fixed maximum image dimension.
-  // Note the max dimension of siftGPU is the maximum dimension of the
-  // first scale in the pyramid(which is the 'first_octave')
-  const double compensation_ratio = options.first_octave < 0 ?
-                              std::pow(2, -options.first_octave) : 1;
+  // Set maximum image dimension.
+  // Note the max dimension of SiftGPU is the maximum dimension of the
+  // first octave in the pyramid (which is the 'first_octave').
+  const double compensation_ratio = std::pow(2, -options.first_octave);
```

Comment:
Hello Johannes, I am a little confused by this line's modification. No image up-sampling happens in the GPU version SIFT when 'first_ocatve' > 1.  In that case, image resizing happens only in ImageResizerThread. So I think the compensation_ratio should be 1  when 'first_ocatve' > 1. Otherwise, image will be downsampled more than we've 
 expected.

Reply:
Yes, I overlooked the fact that SIFT doesn't handle the upsampling. Should be fixed here: https://github.com/colmap/colmap/pull/1642